### PR TITLE
Add support for extends

### LINKED ISSUE #160: Add support for extends
Per #112, grunt-eslint does not currently support `extends`. Please add this support.

### PR DESCRIPTION
**Pull request by ghost**
Closes #160 
image
https://eslint.org/docs/developer-guide/nodejs-api

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#160: Add support for extends](https://issuehunt.io/repos/11402746/issues/160)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

**Comment by sindresorhus** [comment]
It should be exposed as `extends` option, not `baseConfig.extends`.

File: readme.md [hunk 1]
```diff
@@ -56,11 +56,26 @@ grunt.initConfig({
...
 ```
 
+### Base config
+
+```js
+grunt.initConfig({
+	eslint: {
+		options: {
+			baseConfig: {
+				extends: ["eslint-config-shared"]
```
Comment (sindresorhus) [inline readme.md]: single-quotes

Comment (pro-src) [reply readme.md]: Would prefer `'plugin:ava/recommended'` here?

Comment (sindresorhus) [reply readme.md]: Nah, it's just an example.

**Comment by pro-src** [comment]
@sindresorhus If desired, all of the baseConfig options can be hoisted by using ESLint's config schema: https://github.com/sindresorhus/grunt-eslint/compare/master...pro-src:hoist_base_config?expand=1
https://github.com/eslint/eslint/blob/master/conf/config-schema.js

**Comment by pro-src** [comment]
@sindresorhus Don't forget about me :smile:

**Comment by pro-src** [comment]
@sindresorhus,

I know you have 1k other PRs to like at that are far more interesting but I'm bored so I'm going to bug you anyway. :stuck_out_tongue_winking_eye: 

I'm wondering if you had something else in mind for this PR. As the the issue title is confusing to me. Simply because there is already support for the "extends" option. I get what you mean by it should be exposed as "extends" but it has already been exposed as "extends" by ESLint though nested under the documented `baseConfig` option. I've taken a guess at what the expectation is for this, made changes, and suggested some ideas. So, I don't really know what else to do with it. Also, If you just didn't realize that it was already exposed, feel free to kick this out of limbo and into the trash. I don't mind. :smile: 

Thanks

**Comment by sindresorhus** [comment]
Nah, the point is to just make the `extends` option more discoverable and easier to use by making it top-level and documenting it.

--

> I get what you mean by it should be exposed as "extends" but it has already been exposed as "extends" by ESLint though nested under the documented baseConfig option.

Yes, but it's not discoverable.

File: readme.md [hunk 2]
```diff
@@ -99,3 +112,10 @@ Type: `boolean`<br>
 Default: `true`
 
 Fail the build if ESLint found any errors.
+
+### extends
+
+Type: `string` | `array`<br>
+Default: `''`
```
Comment (kevva) [inline readme.md]: Just remove this line, it doesn't really have a default.

File: tasks/eslint.js [hunk 3]
```diff
@@ -9,8 +9,15 @@ module.exports = grunt => {
 			quiet: false,
 			maxWarnings: -1,
 			failOnError: true,
+			extends: false
```
Comment (kevva) [inline tasks/eslint.js]: This is misleading imo. The documentation clearly says it accepts a `string` or an `array`, but here you define it as a `boolean`. I'd remove this line altogether instead.

File: tasks/eslint.js [hunk 4]
```diff
@@ -9,8 +9,15 @@ module.exports = grunt => {
...
 		});
 
+		if (options.extends) {
+			options.baseConfig = Object.assign({
```
Comment (kevva) [inline tasks/eslint.js]: You can use spread here since this module supports Node.js 8 and up.

File: tasks/eslint.js [hunk 5]
```diff
@@ -9,8 +9,15 @@ module.exports = grunt => {
...
+				extends: options.extends
+			}, options.baseConfig);
```
Comment (kevva) [inline tasks/eslint.js]: Should this option override any `extends` in `baseConfig`? If so, the order need to be reversed.

File: readme.md [hunk 6]
```diff
@@ -56,11 +56,24 @@ grunt.initConfig({
...
+		target: ['file.js']
+	}
+});
+```
 
 ## Options
 
 See the [ESLint options](https://eslint.org/docs/developer-guide/nodejs-api#cliengine).
 
+
```
Comment (kevva) [inline readme.md]: Don't add unrelated changes.

**Review by kevva — CHANGES_REQUESTED** [review]

File: readme.md (hunk 2 above)
Comment (pro-src) [reply readme.md]: Sure, I was following the example of `outputFile` which actually defaults to false. Those defaults are not easily accessible and I didn't think it mattered. But the change definitely make sense :+1:

File: tasks/eslint.js (hunk 3 above)
Comment (pro-src) [reply tasks/eslint.js]: Sure, I was following the example of `outputFile` which actually defaults to false. Those defaults are not easily accessible and I didn't think it mattered. But the change definitely make sense :+1:

File: tasks/eslint.js (hunk 5 above)
Comment (pro-src) [reply tasks/eslint.js]: The only common property between the two is the one I've added here. If somebody is using the baseConfig option, they're following ESlint's docs since it's not documented here. IMHO, it makes the most sense as is.

File: readme.md (hunk 6 above)
Comment (pro-src) [reply readme.md]: An extra space :laughing: 
Sorry, didn't realize rofl...

**Comment by pro-src** [comment]
@kevva Thanks for the review. Did you notice https://github.com/sindresorhus/grunt-eslint/compare/master...pro-src:hoist_base_config?expand=1 ?

**Comment by pro-src** [comment]
I visited @sindresorhus patreon page yesterday and seeing your profile image just now, I just can't ignore it. I will not be associated with faggots. I don't know how I missed it before but yeah... You won't be hearing from me anymore....
